lib/backend/watchersyncer/watchercache.go

```diff
@@ -401,3 +407,13 @@ func (wc *watcherCache) markAsValid(resourceKey string) {
 		}
 	}
 }
+
+// sendWaitForDatastore signals to the syncer that this watcherCache is not in-sync if the number of consecutive errors
+// exceeds the error threshold.  See finishResync() for how the watcherCache goes back to in-sync.
+func (wc *watcherCache) sendWaitForDatastore() {
```

Comment:
nit: if this is called `sendWaitForDatastore()` I'd expect it to send a wait for datastore event every time it is called.

I think it would be good to rename this to something like `onError()` or something, which makes it more clear _when_ this function is meant to be called (when we get an error).